ledger-tool/src/main.rs

```diff
@@ -4104,16 +4086,22 @@ fn main() {
                         Box::new(stdout())
                     };
 
-                slot_hash
+                iter.take(num_roots)
+                    .take_while(|slot| *slot <= max_height as u64)
+                    .collect::<Vec<_>>()
                     .into_iter()
                     .rev()
-                    .enumerate()
-                    .for_each(|(i, (slot, hash))| {
-                        if i < num_roots {
-                            output_file
-                                .write_all(format!("{slot:?}: {hash:?}\n").as_bytes())
-                                .expect("failed to write");
-                        }
+                    .for_each(|slot| {
+                        let blockhash = blockstore
+                            .get_slot_entries(slot, 0)
+                            .unwrap()
+                            .last()
+                            .unwrap()
+                            .hash;
+
+                        output_file
```

Comment:
I've mostly seen this written using `writeln!`, rather than `.write_all(format!())`.

Reply:
Fair - initially left it alone as this code didn't change but doubled back and cleaned it up